This patch file is from Qualcomm Technologies, Inc. and is provided under the following license:

Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
SPDX-License-Identifier: BSD-3-Clause-Clear

From 17ed62fdc8df57df82b1e110def8be20c4796e91 Mon Sep 17 00:00:00 2001
From: daofqi <daofqi@qti.qualcomm.com>
Date: Wed, 1 Jul 2026 10:15:25 +0800
Subject: [PATCH] meta-qti-bsp: fix tbb compile issues in perf build on aarch64

[Description]: fix aarch64 perf build failure: libtbb.so ld relocation.
-dangerous relocation: unsupported relocation
-relocation R_AARCH64_ADR_PREL_PG_HI21 against symbol `_ZTVN3tbb6detail2r110user_abortE'.

Signed-off-by: daofqi <daofqi@qti.qualcomm.com>
---
 cmake/compilers/GNU.cmake | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/cmake/compilers/GNU.cmake b/cmake/compilers/GNU.cmake
index 86958e3a..9fd98eea 100644
--- a/cmake/compilers/GNU.cmake
+++ b/cmake/compilers/GNU.cmake
@@ -65,8 +65,10 @@ endif()
 
 # For some reason GCC does not instrument code with Thread Sanitizer when lto is enabled and C linker is used.
 if (NOT TBB_SANITIZE MATCHES "thread")
-    set(TBB_IPO_COMPILE_FLAGS $<$<NOT:$<CONFIG:Debug>>:-flto>)
-    set(TBB_IPO_LINK_FLAGS $<$<NOT:$<CONFIG:Debug>>:-flto>)
+    if (NOT CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64")
+    	set(TBB_IPO_COMPILE_FLAGS $<$<NOT:$<CONFIG:Debug>>:-flto>)
+    	set(TBB_IPO_LINK_FLAGS $<$<NOT:$<CONFIG:Debug>>:-flto>)
+    endif()
 endif()
 
 # TBB malloc settings
-- 
2.34.1

